File: FrontEnd/xego_rider/lib/models/Dto/create_ride_request_dto.dart
class CreateRideRequestDto {
  final String riderId;
  String? driverId;
  int? discountId;
  int? vehicleId;
  final int vehicleTypeId;
  final String status;
  final double startLatitude;
  final double startLongitude;
  final String startAddress;
  final double destinationLatitude;
  final double destinationLongitude;
  final String destinationAddress;
  final double distanceInMeters;
  final DateTime pickupTime;
  final bool isScheduleRide;
  String? cancelledBy;
  String? cancellationReason;
  final String modifiedBy;

  CreateRideRequestDto({
    required this.riderId,
    this.driverId,
    this.discountId,
    this.vehicleId,
    required this.vehicleTypeId,
    required this.status,
    required this.startLatitude,
    required this.startLongitude,
    required this.startAddress,
    required this.destinationLatitude,
    required this.destinationLongitude,
    required this.destinationAddress,
    required this.distanceInMeters,
    required this.pickupTime,
    required this.isScheduleRide,
    this.cancelledBy,
    this.cancellationReason,
    required this.modifiedBy,
  });

  factory CreateRideRequestDto.fromJson(Map<String, dynamic> json) {
    return CreateRideRequestDto(
      riderId: json['riderId'],
      driverId: json['driverId'],
      discountId: json['discountId'],
      vehicleId: json['vehicleId'],
      vehicleTypeId: json['vehicleTypeId'],
      status: json['status'],
      startLatitude: json['startLatitude'],
      startLongitude: json['startLongitude'],
      startAddress: json['startAddress'],
      destinationLatitude: json['destinationLatitude'],
      destinationLongitude: json['destinationLongitude'],
      destinationAddress: json['destinationAddress'],
      distanceInMeters: json['distanceInMeters'],
      pickupTime: DateTime.parse(json['pickupTime']),
      isScheduleRide: json['isScheduleRide'],
      cancelledBy: json['cancelledBy'],
      cancellationReason: json['cancellationReason'],
      modifiedBy: json['modifiedBy'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'riderId': riderId,
      'driverId': driverId,
      'discountId': discountId,
      'vehicleId': vehicleId,
      'vehicleTypeId': vehicleTypeId,
      'status': status,
      'startLatitude': startLatitude,
      'startLongitude': startLongitude,
      'startAddress': startAddress,
      'destinationLatitude': destinationLatitude,
      'destinationLongitude': destinationLongitude,
      'destinationAddress': destinationAddress,
      'distanceInMeters': distanceInMeters,
      'pickupTime': pickupTime.toIso8601String(),
      'isScheduleRide': isScheduleRide,
      'cancelledBy': cancelledBy,
      'cancellationReason': cancellationReason,
      'modifiedBy': modifiedBy,
    };
  }
}

File: BackEnd/XeGo-Microservices/Services/Ride/XeGo.Services.Ride.API/Models/Dto/CreateRideRequestDto.cs
using System.ComponentModel.DataAnnotations;
using XeGo.Shared.Lib.Constants;

namespace XeGo.Services.Ride.API.Models.Dto
{
    public class CreateRideRequestDto
    {
        [Required] public string RiderId { get; set; } = string.Empty!;
        public string? DriverId { get; set; }
        public int? DiscountId { get; set; }
        public int? VehicleId { get; set; }
        [Required] public int VehicleTypeId { get; set; }
        [Required] public string Status { get; set; } = RideStatusConstants.FindingDriver;
        [Required] public double StartLatitude { get; set; }
        [Required] public double StartLongitude { get; set; }
        [Required] public string StartAddress { get; set; } = String.Empty!;
        [Required] public double DestinationLatitude { get; set; }
        [Required] public double DestinationLongitude { get; set; }
        [Required] public string DestinationAddress { get; set; } = String.Empty!;
        [Required] public double DistanceInMeters { get; set; }
        [Required] public string PickupTime { get; set; } = String.Empty!;
        [Required] public bool IsScheduleRide { get; set; }
        public string? CancelledBy { get; set; }
        public string? CancellationReason { get; set; }
        [Required] public string ModifiedBy { get; set; } = String.Empty!;
    }
}
